src/core/useSpriteLoader.tsx

```diff
@@ -1,19 +1,66 @@
+/* eslint react-hooks/exhaustive-deps: 1 */
 import { Texture, TextureLoader } from 'three'
-import { useLoader, useThree } from '@react-three/fiber'
+import { useLoader, useThree, Vector3 } from '@react-three/fiber'
 import { useState } from 'react'
 import * as React from 'react'
 import * as THREE from 'three'
 
+type Size = {
+  w: number
+  h: number
+}
+
+type FrameData = {
+  frame: {
+    x: number
+    y: number
+    w: number
+    h: number
+  }
+  rotated: boolean
+  trimmed: boolean
+  spriteSourceSize: {
+    x: number
+    y: number
+    w: number
+    h: number
+  }
+  sourceSize: Size
+}
+
+type MetaData = {
+  version: string
+  size: {
+    w: number
+    h: number
+  }
+  rows: number
+  columns: number
+  frameWidth: number
+  frameHeight: number
+  scale: string
+}
+
+type SpriteData = {
+  frames: Record<string, FrameData[]> | FrameData[]
+  meta: MetaData
+}
+
+type SpriteMetaDimension = {
+  row: number
+  col: number
+}
+
 // utils
-export const getFirstItem = (param: any): any => {
+export const getFirstItem = (param: FrameData[] | Record<string, FrameData[]>) => {
   if (Array.isArray(param)) {
     return param[0]
   } else if (typeof param === 'object' && param !== null) {
     const keys = Object.keys(param)
 
     return param[keys[0]][0]
   } else {
-    return { w: 0, h: 0 }
+    return { w: 0, h: 0, sourceSize: { w: 0, h: 0 } }
   }
 }

```

Comment:
can't reuse getFirstFrame from SpriteAnimator?

Reply:
i did in https://github.com/pmndrs/drei/pull/2171/commits/a15183296dc433425992b19dfc33fcc0c4aa89a7